Added support for composer v2

boesing opened the pull request:
|    Q          |   A
|-------------- | ------
| BC Break      | no
| New Feature   | yes

boesing committed:
Added support for composer v2

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

michalbundyra commented on .travis.yml:
```diff
@@ -39,6 +39,7 @@ before_install:
 install:
   - travis_retry composer install $COMPOSER_ARGS --ignore-platform-reqs
   - if [[ $LEGACY_DEPS != '' ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
+  - if [[ $DEPS == 'latest' ]]; then travis_retry composer self-update --2 && composer show ; fi
```
see line 45 (46), we already have `composer show`

michalbundyra commented on composer.json:
```diff
@@ -30,11 +30,12 @@
     "require-dev": {
-        "composer/composer": "^1.5.2",
+        "composer/composer": "^1.5.2 || ^2.0@dev",
+        "composer/semver": "^1.5.1 || ^2.0@dev",
```
@boesing probably no longer needed as we have already stable [semver 2.0](https://github.com/composer/semver/releases/tag/2.0.0).

boesing committed:
Remove `composer/semver` dependency

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Changed dev-dependency for `composer/composer`

Add composer 2.0.0 development compatibility and support for upcoming 2.x releases

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>
